verl/trainer/ppo/actor_env.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2024 Bytedance Ltd. and/or its affiliates
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This is the default ActorEnvironment that performs the rollout and actor update.
+"""
+
+import uuid
+import numpy as np
+from copy import deepcopy
+
+
+class ActorEnvironment:
+
+    def __init__(self, config, actor_rollout_wg, reward_fn):
+        self.config = config
+        self.actor_rollout_wg = actor_rollout_wg
+        self.reward_fn = reward_fn
+
+    def step(self, batch, gen_batch, timing_raw):
```

Comment:
Could you provide a high-level abstraction of what step and update should do and how to extend them to new scenarios?

Reply:
Just add some docstring will have great help

Reply:
I just added docstrings for step and update, making the functionality more obvious to users. Extension will come later in a later PR, I hope the current docstrings will be helpful in guiding users.

Reply:
@vermouth1992 Please let me know if we can resolve this conversation and get this part merged. Then I will start the work on part 2.

Reply:
LGTM. @PeterSH6 Could you please comment also?

Reply:
Hi @vermouth1992. It will be hard to develop the example before this PR is merged. If I develop both things at the same time, there will be a lot of merge conflicts to resolve as it will be developed on an old commit, making it less feasible.

This is why part 1 is about allowing extensibility in veRL because then part 2 can work with the latest changes on the main branch.

What do you think about this?

Reply:
@vermouth1992 If it helps, I can import the Search-R1 and adapt it to DuckDuckGo in an example that uses a custom `ActorEnvironment`. This will be an easier feature because I don't need to develop anything that's new.